### ISSUE TITLE
Separate coordinate systems for different coordinates (x,y,...)

### ISSUE THREAD

**Issue by Krzmbrzl** [pgf-tikz/pgf#988]
**Brief outline of the proposed feature**

We can specify the coordinate system for any given point, but it would be nice if optionally we could specify a cs for every coordinate of the point

**Usage example**

```latex
\documentclass{article}

\usepackage{tikz}
\usepackage{pgfplots}

\pgfplotsset{compat=1.17}

\begin{document}
	\begin{tikzpicture}
		\begin{axis}
			\addplot {x};

			% Note the different coordinate systems per coordinate for the second points
			\draw (0,1) -- (axis cs:2, rel axis cs:0.1);
		\end{axis}
	\end{tikzpicture}
\end{document}
```
This example currently fails with
```
tester.tex|14 error| Package PGF Math Error: Unknown function `rel' (in ' rel axis cs:0.1').
```

I used `pgfplots` in my example since here we have the absolute and relative axis cs. There are situations in which it would be useful to provide one coordinate in absolute axis coordinate space but for the other coordinate it is not that important where exactly the point ends up being so a relative coordinate would be more convenient to use (especially if the axis has a range that makes using absolute axis coordinate values a bit of a pain (e.g. `1.3451876E-24`)).
In my specific case I want to use the `spy` library to magnify a part of my plot. I want to specify the `x` coordinate exactly using absolute axis cs but since my y-axis scale is purposefully hidden, it would be tedious to show the scale, specify the y coord and then hide the scale again. Thus I would like to specify that the point should be at 10% total height.

~~A workaround that am going to use is to define 2 points using the absolute and the relative axis cs respectively and then combine their x and y coordinates using `-|` as explained in https://tex.stackexchange.com/questions/63560/how-to-use-x-coordinate-of-a-point-and-y-coordinate-of-other-point to define a third point that then (hopefully) has the coordinates I want it to have.
It'd be nice if I could do that with defining only a single point though.~~
EDIT: The described workaround doesn't seem to work either

**Comment by hmenke** [comment]
This already exists
```tex
% |- takes the x-coordinate of the first and the y-coordinate of the second.
\draw (0,1) -- ({axis cs:2,0} |- {rel axis cs:0,0.1});
```

**Comment by Krzmbrzl** [comment]
Thanks for the fast response! Works like charm :+1: 

Is this in the manual? I did not find it specified in the section on coordinate systems. Is this perhaps a newer feature? My manual is a bit outdated (for 3.0.1a)

**Comment by hmenke** [comment]
![test](https://user-images.githubusercontent.com/1903556/106357476-86f3a480-6306-11eb-9d08-6bbb58dd6c7c.png)

**Comment by Krzmbrzl** [comment]
I was actually more referring to the usage of this feature to specify coordinates in different coordinate systems within a singke point :point_up:

**Comment by muzimuzhi** [comment]
In syntax `(<p> |- <q>)`, the only requirement is that both `<p>` and `<q>` are coordinates. They can be in any forms, including in different coordinate systems.

**Comment by Krzmbrzl** [comment]
Okay let me rephrase my original intention:
It might be a good idea to add this trick of using `|-` to specify x and  coordinates in a different coordinate system to the section in the manual that describes what coordinate systems are and how they can be used.

(This is a suggestion rather than a question)

**Comment by hmenke** [comment]
The manual contains several tutorials and an extensive reference manual.  Users are expected to work through the tutorials and then consult the reference manual to piece together their own solutions.  For questions about usage of PGF/TikZ please head over to https://tex.stackexchange.com/
